### DISCUSSION TITLE
Interest in adding paging support?

### DISCUSSION THREAD

**Discussion by jheddings** [ramnes/notion-sdk-py#50] [Ideas]
I've been working on a basic integration that pulls from databases.  I ended up writing a paginating iterator to loop over results.  I'm sure it could be improved, but I was wondering if there would be any interest including this in the SDK as a helper class?

It's an early start, but here's an example of what I'm working on:
https://github.com/jheddings/notion-sdk-py/blob/pager/notion_client/pager.py

**Comment by ramnes** [comment]
Yeah, definitely. :)

It's funny because I have been discussing the exact same feature on Slack an hour ago or so! I was saying that we're in a weird situation because I started this library as a 1:1 rewrite of Notion's library, but yet we all want more features. You can read my thoughts on the matter here: https://github.com/ramnes/notion-sdk-py/issues/9#issuecomment-841688799

For this kind of helper in particular, I'd say we could simply add them in helpers.py until we land a better code structure.

**Reply by jheddings** [reply]
Cool!  I can move things around and submit a PR if that makes the most sense.  I'm only working with databases at the moment, so I have not extended this to other result sets.

FWIW, keeping the core API implementation close to the reference is a great idea.  The reality is that we are all dealing with the same problems using the core API in an app.  In fact, the "dynamic objects" problem is what led me to your SDK (which fired right up).  I've been a `notion-py` user & contributor for a while, which makes higher-level access almost "too magical."  Getting the balance right is tough!

**Reply by ramnes** [reply]
Yes, please submit a PR, although I would be looking for something applicable to all endpoints. :)

And yes, I do think that notion-py has too much black magic, but it's always a lot of fun when you know how it works behind the scenes! I think I'll try to restrain myself a tad earlier though.
